pyuvdata/uvdata/uvdata.py

```diff
@@ -3192,6 +3409,15 @@ def __add__(
         else:
             fnew_inds = []
 
+        # if channel width is an array and there's any overlap in freqs,
+        # check extra params
+        if this.future_array_shapes or this.Nspws > 1:
```

Comment:
I think the latter part of this should be `or this.flex_spw == True` rather than `or this.Nspws > 1`.

Reply:
yep, found that one just now too.